source/server/http_route_manager_impl.cc

```diff
@@ -0,0 +1,34 @@
+#include "server/http_route_manager_impl.h"
+
+namespace Envoy {
+namespace Server {
+
+std::vector<Router::RouteConfigProviderSharedPtr> HttpRouteManagerImpl::routeConfigProviders() {
+  std::vector<Router::RouteConfigProviderSharedPtr> ret;
+  ret.reserve(route_config_providers_.size());
+  for (const auto& element : route_config_providers_) {
+    ret.push_back(element.second.lock());
```

Comment:
@mattklein123 is it possible here that between the time that the last shared_ptr for the provider gets reset, and the destructor for the provider is called (thus cleaning the entry in the manager), this iteration happens?

In other words, should I lock, check for nullptr, and then push into the vector only is the shared_ptr is not null?